lib/src/main/kotlin/org/didcommx/didcomm/model/PackPlaintextParams.kt

```diff
@@ -55,6 +72,26 @@ data class PackPlaintextParams(
          *
          * @return Pack Plaintext Message Parameters
          */
-        fun build() = PackPlaintextParams(this)
+        fun build(): PackPlaintextParams {
+            val fromPriorIssuerKid = this.fromPriorIssuerKid
+
+            if (fromPriorIssuerKid != null && (!isDID(fromPriorIssuerKid) || !isDIDFragment(fromPriorIssuerKid)))
+                throw DIDCommIllegalArgumentException(fromPriorIssuerKid)
+
+            if (message.fromPrior != null) {
```

Comment:
Looks like a common validation logic (plaintext/signed/encrypted). Makes sense to move to one place.